app/services/org/create_created_plan_service.rb

```diff
@@ -6,24 +6,54 @@ def call(org = nil)
 
         orgs.each do |org|
           OrgDateRangeable.split_months_from_creation(org) do |start_date, end_date|
-            create_count_for_date(start_date: start_date, end_date: end_date, org: org)  
+            create_or_update(start_date: start_date, end_date: end_date, org: org)
           end
         end
       end
 
       private
 
+      def users(org)
+        User.where('users.org_id = ?', org.id)
+      end
+
+      def plans(start_date:, end_date: )
+        Plan.where('plans.created_at >= ? AND plans.created_at <= ?', start_date, end_date)
+      end
+
+      def creator_admon
+        Role.with_access_flags(:creator, :administrator)
+      end
+
       def count_plans(start_date: , end_date: , org:)
-        users = User.where('users.org_id = ?', org.id)
-        plans = Plan.where('plans.created_at >= ? AND plans.created_at <= ?', start_date, end_date)
-        creator_admon = Role.with_access_flags(:creator, :administrator)
+        users = users(org)
+        plans = plans(start_date: start_date, end_date: end_date)
 
         Role.joins([:plan, :user]).merge(creator_admon).merge(users).merge(plans).select(:plan_id).distinct.count
       end
 
-      def create_count_for_date(start_date:, end_date:, org:)
+      def by_template(start_date:, end_date:, org:)
+        users = users(org)
+        plans = plans(start_date: start_date, end_date: end_date)
+        roleable_plan_ids = Role.joins([:plan, :user]).merge(creator_admon).merge(users).merge(plans).select(:plan_id).distinct
+        template_counts = Plan.where(id: roleable_plan_ids).group(:"template_id").count
```

Comment:
template_id doesn't need to be in quotes

Reply:
ops! you're right it is pointless to symbolize that way